WebEngine/JsonTransport.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WEB_ENGINE_TRANSPORT_H_
+#define WEB_ENGINE_TRANSPORT_H_
+
+#include <QObject>
+#include <QWebChannelAbstractTransport>
+#include <QWebSocket>
+#include <memory>
+
+#include "WebEngine/DeleteLaterDeleter.h"
+
+namespace WebEngine {
+
+/* Json-based transport for QtWebChannel
+
+   A transport in the QtWebChannel framework is the link
+   between to C++ WebChannel object and the websocket talking
+   to the JavaScript part.
+
+   This is a very basic implementation and also the default.
+   It encodes all messages as JSON. The default JavaScript
+   channel implementation (qwebchannel.js) is doing the same.
+   So it can't be easily swapped out without rewriting the JS
```

Comment:
I don't really understand this sentence. Why is that that if the JavaScript channel implementation does the same as this implementation, then "it" (what is it?) can't be easily swapped out?

Reply:
Good point. Rephrasing...

Reply:
Done.